docs/docsite/_ext/build_context.py

```diff
@@ -0,0 +1,110 @@
+"""Sphinx extension for setting up the build settings."""
+
+import subprocess
+from dataclasses import dataclass
+from functools import cache
+from pathlib import Path
+from tomllib import loads as parse_toml_string
```

Comment:
Do we still need to support Python 3.10?

Reply:
Could add an extra optional dependency if needed. Though given that 3.11 is hardcoded in the automation, I'd not bother.

Reply:
The `stable-2.13` is the only branch where we'd need to support Python 3.10. Initially I was thinking we'd want to merge this back to every branch for consistency. However it that requires extra dependencies for this extension I don't think it is worth it.

We can still declare the `stable-2.13` branch in the toml file for completeness. And it would also be good to document the process to EOL docs in the README. We could just mention the "legacy" eol flag on those older branches. It's already set there anyway and there's no reason to ever revert it.